src/App.tsx

```diff
@@ -1,14 +1,66 @@
 /* eslint-disable max-len */
-import React from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
+import { useEffect, useState } from 'react';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loadingTodos, setLoadingTodos] = useState(false);
+  const [query, setQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [status, setStatus] = useState('');
+
+  const handleStatusChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setStatus(e.target.value);
+  };
+
+  const filteredTodos = todos.filter(todo =>
+    todo.title.toLowerCase().includes(query),
+  );
+
+  const filteredTodosByStatus = filteredTodos.filter(todo => {
+    if (status === 'all') {
+      return true;
+    }
+
+    if (status === 'active') {
+      return todo.completed === false;
+    }
+
+    if (status === 'completed') {
+      return todo.completed === true;
+    }
+
+    return true;
+  });
+
+  const handleAllTodos = (input: string) => {
+    setQuery(input.toLowerCase());
```

Comment:
it means the user can't type big letters which might be a bit confusing

It's better to do `query.toLowerCase()` inside filter to get `filteredTodos` instead of doing it here